hibernate-core/src/main/java/org/hibernate/boot/model/internal/ClassPropertyHolder.java

```diff
@@ -274,6 +275,9 @@ public static void addPropertyToMappedSuperclass(
 			ClassDetails declaringClass,
 			MetadataBuildingContext context) {
 		final MappedSuperclass superclass = context.getMetadataCollector().getMappedSuperclass( declaringClass.toJavaClass() );
+		if ( superclass == null ) {
+			throw new HibernateException( "Mapped superclass is null for declaring type " + declaringClass );
```

Comment:
This message is slightly better than an NPE, but it doesn't  really tell the user what they did wrong, nor what they should do to fix it.

Also, shouldn't it be a `MappingException`?

Reply:
> I think we might want to understand why the `superclass` is null, too, since looking at the code-path that leads here (from the Jira's stack trace) it looks like we're processing an embeddable that extends from a `@MappedSuperclass`. I fear that we might be processing the embeddable before its parent, I remember fixing something similar for https://hibernate.atlassian.net/browse/HHH-18103.

One of the issue's comments states "the superclass of the embedded id was not part of the mapped resources that Spring registers with Hibernate", so it seems like this was an error in Spring
@gavinking This is also why I chose to throw a more generic HibernateException, since we may not be sure what the cause of the missing MappedSuperclass is?